### ISSUE TITLE
json creation when converting from nnUNetV2 to BIDS 

### ISSUE THREAD

**Issue by tzebre** [ivadomed/utilities#20]
The BIDS format contains a json file for each nifti file in the dataset. 
The json file contains 2 information:
- Author 
- Date of creation 
```
{
    "Author": "Sandrine Bedard",
    "Date": "2021-06-08 15:05:54"
}
```

In the [nnunetV2_to_bids.py](https://github.com/ivadomed/utilities/blob/a1938d6d26c19e59fea5c2181f46b52589cdac92/dataset_conversion/nnunetV2_to_bids.py#L67) script, we create a BIDS dataset from an nnUNet dataset. 
We have 2 choices in the case of script-generated BIDS. 
The script create the json file: 
```
{
    "Author": "nnUNetV2_to_bids.py from nnUNet dataset NAME OF DATASET",
    "Date": "2023-06-27 14:00:51"
}
```
Or the reviewer creates the JSON when they manually review the file. 

Do you have any ideas on what choice we need to implement?

**Comment by jcohenadad** [comment]
Better to have it done automatically, no? Humans make mistake

**Comment by tzebre** [comment]
Yes, of course.

In the case the nnUNet dataset is the result of an nnUNet prediction:  
- Letting the reviewer create the json ensures that the file will be reviewed before the BIDS dataset is complete (if we consider that without json file the BIDS dataset is incomplete). 
- Automatically creating the JSON means that we consider the label from the nnUNet as truth. But if we say in the documentation that the label must be checked before the conversion script, this problem no longer exists.

**Comment by jcohenadad** [comment]
> Automatically creating the JSON means that we consider the label from the nnUNet as truth. But if we say in the documentation that the label must be checked before the conversion script, this problem no longer exists.

Not necessarily. It just means that the label was created by nnUNet. Then, if the label is reviewed by a human, we should concatenate the reviewer(s) to the "nnUNet" entry. 

Ex: we could append another structure below:
```
{
    "Author": "nnUNetV2_to_bids.py from nnUNet dataset NAME OF DATASET",
    "Date": "2023-06-27 14:00:51"
}

{
    "Author": "Donald Trump",
    "Date": "2023-06-28 12:30:01"
}
```

**Comment by tzebre** [comment]
> Not necessarily. It just means that the label was created by nnUNet. Then, if the label is reviewed by a human, we should concatenate the reviewer(s) to the "nnUNet" entry.
> 
> Ex: we could append another structure below:
> 
> ```
> {
>     "Author": "nnUNetV2_to_bids.py from nnUNet dataset NAME OF DATASET",
>     "Date": "2023-06-27 14:00:51"
> }
> 
> {
>     "Author": "Donald Trump",
>     "Date": "2023-06-28 12:30:01"
> }
> ```

Yes really good idea.

**Comment by jcohenadad** [comment]
Tagging @valosekj @sandrinebedard because they are involved in the "neuropoly good-practice SOP for label creation", so that idea could be implemented in our SOP

**Comment by valosekj** [comment]
Thank you for coming up with this relevant question, @tzebre!

A bit of context: currently, we store JSON sidecars for manually corrected label files (SC seg, disc labeling, etc.) to track the information about who performed manual corrections. If multiple users did corrections, we store all of them (see [here](https://github.com/spinalcordtoolbox/manual-correction/issues/34#issuecomment-1494663834)). 
But in the case when label files are generated purely automatically (for example, SC seg generated by [`sct_deepseg_sc`](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/master/spinalcordtoolbox/scripts/sct_deepseg_sc.py)), we either have no JSON sidecar, or we use empty one during dataset BIDSification.

I think this issue opens a broader question: do we want to generate JSON sidecars for automatically generated files? I would vote for it. We could store the algorithm (e.g., propseg vs deepseg_sc vs nnUNet, etc.) and its version.
I'll put this point on the agenda for the next SCT dev meeting.

**Comment by jcohenadad** [comment]
> I think this issue opens a broader question: do we want to generate JSON sidecars for automatically generated files?

Yes, definitely. This is extremely important for tracking provenance. Storing the algorithm and the version is important.

**Comment by valosekj** [comment]
> > I think this issue opens a broader question: do we want to generate JSON sidecars for automatically generated files?
> 
> Yes, definitely. This is extremely important for tracking provenance. Storing the algorithm and the version is important.

Okay! I found that SCT already has an open issue (https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3394) about this topic --> discussion redirected there.
